packages/api/models/rules.js

```diff
@@ -228,39 +239,51 @@ class Rule extends Manager {
       newRuleItem.state = 'ENABLED';
     }
 
-    const payload = await Rule.buildPayload(newRuleItem);
-    switch (newRuleItem.rule.type) {
-    case 'onetime': {
-      await invoke(process.env.invoke, payload);
-      break;
-    }
-    case 'scheduled': {
-      await this.addRule(newRuleItem, payload);
-      break;
-    }
-    case 'kinesis': {
-      const ruleArns = await this.addKinesisEventSources(newRuleItem);
-      newRuleItem = this.updateKinesisRuleArns(newRuleItem, ruleArns);
-      break;
-    }
-    case 'sns': {
-      if (newRuleItem.state === 'ENABLED') {
-        const snsSubscriptionArn = await this.addSnsTrigger(newRuleItem);
-        newRuleItem = this.updateSnsRuleArn(newRuleItem, snsSubscriptionArn);
-      }
-      break;
-    }
-    case 'sqs':
-      newRuleItem = await this.validateAndUpdateSqsRule(newRuleItem);
-      break;
-    default:
-      throw new Error(`Rule type \'${newRuleItem.rule.type}\' not supported.`);
-    }
+    newRuleItem.createdAt = Date.now();
+    newRuleItem.updatedAt = Date.now();
+
+    // Validate rule before kicking off workflows or adding event source mappings
+    await this.constructor.recordIsValid(newRuleItem, this.schema, this.removeAdditional);
+
+    newRuleItem = await this.createRuleTrigger(newRuleItem);
 
     // save
     return super.create(newRuleItem);
   }
 
+  async createRuleTrigger(ruleItem) {
+    let newRuleItem = cloneDeep(ruleItem);
+
+    const payload = await Rule.buildPayload(newRuleItem);
+    switch (newRuleItem.rule.type) {
+      case 'onetime': {
```

Comment:
Weirdly, ESlint seems to think that case statements should not be indented more than the switch statement.